app/src/org/commcare/activities/DispatchActivity.java

```diff
@@ -66,6 +68,11 @@ protected void onCreate(Bundle savedInstanceState) {
         if (savedInstanceState != null) {
             shortcutExtraWasConsumed = savedInstanceState.getBoolean(EXTRA_CONSUMED_KEY);
         }
+
+        // Note: It is important that this check happens in onCreate, and not at the time that
+        // the value is actually used, because we only want to check for changed app files on a
+        // new creation of the DispatchActivity, not every time it resumes
+        shouldCheckForLocalAppFilesChange = CommCareApplication._().isConsumerApp();
```

Comment:
Good call, added a protection for that. One day I will catch rotation errors for myself :(